### PR TITLE
FAM-170 restuctured API to be more friendly

### PR DESCRIPTION
**Pull request by maratkomarov**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,8 @@
+.*
```
Comment (recipe) [inline .gitignore]: @maratkomarov what is the rationale of using negation pattern here?

File: tests/large/AcceptanceTest.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
+<?php
+
+use \Fatmouse\Fatmouse;
```
Comment (recipe) [inline tests/large/AcceptanceTest.php]: @maratkomarov we should not prefix namespaces with `\` in use statements.

File: tests/large/AcceptanceTest.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+class AcceptanceTest extends PHPUnit_Framework_TestCase
+{
+    private static $fatmouse;
+
+    private function getFatmouse()
+    {
+        if (! self::$fatmouse) {
+            
+            self::$fatmouse = new Fatmouse('amqp://localhost');
+            //self::$fatmouse = new Fatmouse('amqp://localhost');
```
Comment (recipe) [inline tests/large/AcceptanceTest.php]: @maratkomarov ?

File: tests/large/AcceptanceTest.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        try {
+            $task = $r->wait();
+            $reg = $task->getResult();
+
+            $this->assertInstanceOf(Tasks\RegisterServer::class, $task);
+            $this->assertInstanceOf(Types\RegisteredServer::class, $reg);
+            $this->assertEquals($reg->serverId, $serverId);
+        } 
+        finally {
```
Comment (recipe) [inline tests/large/AcceptanceTest.php]: @maratkomarov finally should be on the same line `} finally {`

File: Fatmouse/Workflows/Init.php [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * {@inheritdoc}
+ */
+class Init extends Workflow
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @method \Fatmouse\Types\InitializedServer getResult()
```
Comment (recipe) [inline Fatmouse/Workflows/Init.php]: @maratkomarov @method definition should be in the phpdoc above class definition

File: Fatmouse/Workflows/Orchestration.php [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * {@inheritdoc}
+ */
+class Orchestration extends Workflow
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @method \Fatmouse\Types\Orchestration\ExecutedAction[] getResult()
```
Comment (recipe) [inline Fatmouse/Workflows/Orchestration.php]: @maratkomarov the same

File: Fatmouse/Tasks/RegisterServer.php [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * {@inheritdoc}
+ */
+class RegisterServer extends Task
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @method \Fatmouse\Types\RegisteredServer getResult()
```
Comment (recipe) [inline Fatmouse/Tasks/RegisterServer.php]: @maratkomarov the same

File: Fatmouse/Tasks/DeregisterServer.php [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * {@inheritdoc}
+ */
+class DeregisterServer extends Task
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @method void getResult()
```
Comment (recipe) [inline Fatmouse/Tasks/DeregisterServer.php]: @maratkomarov the same

File: Fatmouse/Proxies/Workflows.php [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+     * Orchestrate Scalr event.
+     *
+     * FatMouse will execute scripts associated with a given event on their targets.
+     *
+     * @param \Fatmouse\Types\Orchestration\OrchestrationParams $orchestration Scalr
+     *                                                                         orchestration to run.
+     *
+     * @return \Fatmouse\AsyncResult Result future.
+     */
+    public function orchestration(\Fatmouse\Types\Orchestration\OrchestrationParams $orchestration)
```
Comment (recipe) [inline Fatmouse/Proxies/Workflows.php]: @maratkomarov why don't you `use Fatmouse\Types\Orchestration\OrchestrationParams`. It's a good practice to use `use` statements for all used classes.

File: .gitignore (hunk 1 above)
Comment (maratkomarov) [inline .gitignore]: @recipe don't include all hidden files by default, and enable only that we really need.

File: Fatmouse/EventName.php [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+ */
+class EventName
+{
+    const SERVER_REBOOTED = 'ServerRebooted';
+    const SERVER_REBOOTING = 'ServerRebooting';
+    const TASK_FINISHED = 'TaskFinished';
+    const WORKFLOW_FINISHED = 'WorkflowFinished';
+
+    private static $NAME_TO_PAYLOAD_CLASS = [
+        self::SERVER_REBOOTED => Fatmouse\Events\ServerRebooted::class,
```
Comment (recipe) [inline Fatmouse/EventName.php]: @maratkomarov you could also `use Fatmouse\Events;` and then referencing lile `Events\ServerRebooted::class`

File: Fatmouse/AsyncResult.php [hunk 11]
```diff
@@ -3,99 +3,181 @@
...
+    /**
+     * Called by self::newFromCeleryMessage() factory to construct this object.
+     *
+     * @param \PhpAmqpLib\Connection\AMQPStreamConnection $connection AMQP connection.
+     * @param \stdClass                                   $celeryMsg  Decoded Celery result message.
+     *
+     * @return void
+     */
+    private function initFromCeleryMessage(
+        \PhpAmqpLib\Connection\AMQPStreamConnection $connection,
```
Comment (recipe) [inline Fatmouse/AsyncResult.php]: @maratkomarov `use PhpAmqpLib\Connection\AMQPStreamConnection;` then reference `AMQPStreamConnection`

**Pull request merged**